### PR TITLE
ENH: Feature selective covar correction

### PR DESCRIPTION
**Pull request by itellaetxe**
## Summary
Adds a new option for choosing which group to base the covariate correction on, `--covcorr_mode`, which can take 3 values:
- `cn` -> Only control subjects are used to compute the covariate correction `beta` coefficients.
- `all` -> The whole population is used to compute the covariate correction `beta` coeffs.
- `each` -> A set of `beta` coeffs. for covariate correction is computed for each clinical group.

### Key Changes
- ENH: Adds covariate correction modes in 3 pipelines: `factor_correlation`, `clinical_groups`, and `clinical_classify`.
- FIX: In `ui` module tests, changes sex covariates to numerical because covariate correction does **not** support string covariates for now.

### Other Changes
- STYLE: Corrects several typos
- ENH: Explicitly set the `required` flag to `False` in all optional CLI arguments.

### CODE REVIEW

File: src/ageml/messages.py [hunk 1]
```diff
@@ -73,6 +73,14 @@
     "https://github.com/compneurobilbao/ageml/blob/main/docs/input_file_specification.md#clinical-file"
 )
 
+covcorr_mode_long_description = (
```
Comment (JGarciaCondado) [inline src/ageml/messages.py:76]: Missing description of what is the default if covariates files given but this flag is not.

File: src/ageml/ui.py [hunk 2]
```diff
@@ -365,7 +366,11 @@ def load_covariates(self, required=False):
             self.args.covar_name = self.args.covar_name.lower()
             if self.args.covar_name not in df:
                 raise KeyError("Covariate column %s not found in covariates file." % self.args.covar_name)
-     
+
+        # Check if covariate correction mode and set the flag
```
Comment (JGarciaCondado) [inline src/ageml/ui.py:370]: what if you just input a covariate file. Does it still do covariate correction? it should do it with cn implicitly

**Review by JGarciaCondado — COMMENTED** [review]
Plase address the two comments. Rest looks good to me.

File: src/ageml/messages.py (hunk 1 above)
Comment (itellaetxe) [reply src/ageml/messages.py:76]: True, just added it.

File: src/ageml/ui.py (hunk 2 above)
Comment (itellaetxe) [reply src/ageml/ui.py:370]: `self.flags["covcorr_mode"]` is set to `'cn'`, in the `Interface` initialization, so if the covariate file is provided with no covariate correction mode specified, the mode is `'cn'`.

**Pull request merged**